### PR TITLE
Fixup head when enabled

### PR DESCRIPTION
**Pull request by agx**
An currently disabled head doesn't report a scale or mode. Detect
that case and fill in reasonable defaults.

Closes: #1

### CODE REVIEW

File: main.c [hunk 1]
```diff
@@ -430,9 +430,36 @@ static bool parse_mode(const char *value, int *width, int *height,
 	return true;
 }
 
+
+static void fixup_disabled_head(struct randr_head *head)
+{
```
Comment (emersion) [inline main.c]: Style: brace should be on the previous line

File: main.c [hunk 2]
```diff
@@ -430,9 +430,36 @@ static bool parse_mode(const char *value, int *width, int *height,
...
+	struct randr_mode *mode;
```
Comment (emersion) [inline main.c]: Style: this should be declared right before being used

File: main.c [hunk 3]
```diff
@@ -430,9 +430,36 @@ static bool parse_mode(const char *value, int *width, int *height,
 	return true;
 }
 
+
```
Comment (emersion) [inline main.c]: Style: extra newline here

File: main.c [hunk 4]
```diff
@@ -430,9 +430,36 @@ static bool parse_mode(const char *value, int *width, int *height,
...
+static void fixup_disabled_head(struct randr_head *head)
+{
+	struct randr_mode *mode;
+
+	if (head->scale <= 0.0) {
+		head->scale = 1.0;
```
Comment (emersion) [inline main.c]: Maybe we can just set scale to 1.0 when initializing the `randr_head` instead?

File: main.c [hunk 5]
```diff
@@ -430,9 +430,36 @@ static bool parse_mode(const char *value, int *width, int *height,
...
+	}
+
+	if (!head->mode && head->custom_mode.refresh == 0 &&
+	    head->custom_mode.width == 0 && head->custom_mode.height == 0) {
```
Comment (emersion) [inline main.c]: Style: instead of spaces, we use two tabs

File: main.c [hunk 6]
```diff
@@ -430,9 +430,36 @@ static bool parse_mode(const char *value, int *width, int *height,
...
+		wl_list_for_each(mode, &head->modes, link) {
+			if (mode->preferred) {
+				head->mode = mode;
+				return;
+			}
+		}
+		/* Pick first element if when there's no preferred mode */
+		if (!wl_list_empty(&head->modes)) {
+			head->mode = wl_container_of(head->modes.next,
+						     mode, link);
```
Comment (emersion) [inline main.c]: Style: one tab instead of spaces

File: main.c [hunk 7]
```diff
@@ -430,9 +430,36 @@ static bool parse_mode(const char *value, int *width, int *height,
...
+		}
+	}
+}
+
 static bool parse_output_arg(struct randr_head *head,
 		const char *name, const char *value) {
 	if (strcmp(name, "on") == 0) {
+		if (!head->enabled)
```
Comment (emersion) [inline main.c]: Style: always use braces

**Review by emersion — CHANGES_REQUESTED** [review]
Thanks for the PR! Mostly style issues

File: main.c (hunk 4 above)
Comment (agx) [reply main.c]: yeah, I actually thought about that one when looking at the code and then forgot ;)

File: main.c (hunk 6 above)
Comment (agx) [reply main.c]: fixed

**Comment by agx** [comment]
Fixed now. Is there a clang-format file somewhere for this style?

**Comment by emersion** [comment]
Thanks!

>Fixed now. Is there a clang-format file somewhere for this style?

I have a [WIP clang-format file](https://sr.ht/nZh1.clangformat) that almost gives the wlroots style. It's adding one extra indent level unfortunately. I'd like to send an upstream clang-format patch to fix this.

**Pull request merged**